traversal/procedure/ProcedureVertex.java

```diff
@@ -643,4 +670,87 @@ public ProcedureVertex.Type asType() {
             return this;
         }
     }
+
+    public static class Value extends ProcedureVertex<ValueVertex<?>, Properties.Value> {
+        Value(Identifier identifier) {
+            super(identifier);
+        }
+
+        @Override
+        protected Properties.Value newProperties() {
+            return new Properties.Value();
+        }
+
+        @Override
+        public <ORDER extends Order> Forwardable<? extends ValueVertex<?>, ORDER> iterator(
+                GraphManager graphMgr, Traversal.Parameters parameters, ORDER order, boolean forceValueSort
+        ) {
+            assert iterate(ins()).noneMatch(ProcedureEdge::isInput);
+            ValueVertexImpl<?> tmpResult;
+            try {
+                tmpResult = props().expression().evaluateToVertex(new HashMap<>()); // no inputs
+            } catch (TypeDBException | TypeDBCheckedException e) {
+                throw evaluationError(new HashMap<>(), e);
+            }
+            ValueVertexImpl<?> result = tmpResult;
+            if (iterate(props().predicates()).allMatch(predicate -> checkPredicate(predicate, result, parameters.getValues(id().asVariable(), predicate)))) {
+                return iterateSorted(order, result);
+            } else return iterateSorted(order);
+        }
+
+        @Override
+        public boolean isValue() {
+            return true;
+        }
+
+        @Override
+        public Value asValue() {
+            return this;
+        }
+
+        @Override
+        public Forwardable<Vertex<?, ?>, Order.Asc> intersect(Map<ProcedureEdge<?, ?>, Forwardable<Vertex<?, ?>, Order.Asc>> iterators, Traversal.Parameters parameters, Order.Asc order) {
```

Comment:
Value vertex implementation of intersect.
Takes the assignment inputs, makes an input map, evaluates the value
Collects the predicate inputs, evaluates each against the result of the evaluation